src/server/util/file_io_adaptor.cc

```diff
@@ -0,0 +1,124 @@
+/** Copyright 2020-2021 Alibaba Group Holding Limited.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#include "server/util/file_io_adaptor.h"
+#include <vector>
+
+#include "arrow/filesystem/filesystem.h"
+#include "basic/ds/arrow_utils.h"
```

Comment:
Please move the arrow related macros to a new file `common/util/arrow.h` then we could avoid include `basic/ds/arrow_utils.h` in vineyardd.